emark/message.py

```diff
@@ -129,8 +129,28 @@ def update_url_params(self, url, **params):
 
     def inject_utm_params(self, md, **utm):
         for url in INLINE_LINK_RE.findall(md):
+            try:
+                url_parts = parse.urlparse(url)
+                if url_parts.scheme and url_parts.scheme not in [
```

Comment:
You might want to do a `.lower()` and add a test for an ALLCAPS URL.

Reply:
The end result becomes funny since we are doing an injection:
https://github.com/voiio/emark/pull/94/commits/111b561c423cb0623062dd8fc094624d9d15a1af

We can modify the whole URL to become lowercase, not sure if I want that.